tests/test_core.py

```diff
@@ -169,9 +178,18 @@ def test_executor():
             """Test executor."""
             call_count.append('call')
 
+        @asyncio.coroutine
+        def wait_finish_callback():
+            """Wait until all stuff is sheduled."""
+            while len(call_count) < 40:
+                yield from asyncio.sleep(0)
+
         for i in range(40):
             self.hass.add_job(test_executor)
 
+        run_coroutine_threadsafe(
```

Comment:
I would expect 2 sleeps to be enough for this test. One to get all the async_add_job tasks called, another one to do the tasks scheduled by those tasks.